replica/database.cc

```diff
@@ -415,6 +420,18 @@ void database::on_tp_timer() {
     _tp_listener->reset();
 }
 
+void database::update_toppartitions_listener() {
+    dblog.info("Updating toppartitions listener");
+
+    if (!_tp_listener->set_sampling_probability(_cfg.persistent_toppartitions_sampling_probability())) {
+        dblog.error("Failed to apply persistent toppartitions changes. Sampling probability must be in a [0,1] range");
```

Comment:
Please also log the sampling probability, so we know what the bad value is.

Reply:
Done.